crates/chain/src/descriptor_ext.rs

```diff
@@ -21,9 +21,9 @@ pub trait DescriptorExt {
     /// Panics if the descriptor wildcard is hardened.
     fn dust_value(&self) -> u64;
 
-    /// Returns the descriptor id, calculated as the sha256 of the descriptor, checksum not
+    /// Returns the keychain id, calculated as the sha256 of the descriptor, checksum not
     /// included.
-    fn descriptor_id(&self) -> DescriptorId;
+    fn keychain_id(&self) -> KeychainId;
```

Comment:
Let's have this return an `Option`, non-wildcard descriptors are not keychains.

Reply:
In my latest push I removed the restriction against using non-wildcard descriptors so this function will never fail and doesn't need an `Option`. See updated docs.